enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,451 @@
+        # Needs to be unique per container
+        - --metrics-address=:7980
+        # Filter from aws dev account
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+```
+
+Note the use of multiple ExternalDNS containers within the same pod int he above deployment. Metrics listening ports would have
+to be configured on a per-container basis to avoid collisions.
+
+Metrics will be exposed via an insecure channel by default.
+On OpenShift, externalDNS instances can be deployed with a reverse proxy sidecar, such as kube-rbac-proxy, to provide a secure means of metrics scraping.
+
+### Risks and Mitigations
+
+Cluster administrators may not want to give users the ability to create arbitrary DNS records in an external provider when they grant users permission to create routes.
+ExternalDNS could observe a sub-field of a route resource when determining whether or not to publish records to the upstream provider.
+Cluster administrators could restrict a users ability to set this route sub-field, using a mechanism similar to that for route custom hosts.
+
+Similarly, cluster administrators may want to restrict the ability for users to specify the ExternalDNS
+[DNS target override annotation](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/openshift.md#prepare-router_canonical_hostname-in-defaultrouter-deployment).
+on route resources, since the override annotation could give users the ability to create DNS records for alternative domains.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. If an OpenShift route is exposed by multiple Ingress Controllers, should ExternalDNS publish multiple records?
+1. Can more than one ExternalDNS instance (ie. one for a public zone and one for a private zone) be ran using a single deployment?
+1. Similar to the above, would it be sufficient for a user to create multiple ExternalDNS Operator Custom Resources should they desire separate deployments
+for cross-zone ExternalDNS functionality? Note that there is an [upstream issue](https://github.com/kubernetes-sigs/external-dns/issues/1961) focused on the
+addition of a CRD for configuring a single instance of ExternalDNS to serve multiple providers and zones simultaneously.
+1. Will the wildcard DNS record created for the Default Ingress Controller by the Ingress Operator conflict with how end-users want to use ExternalDNS on OpenShift?
+1. Should the ExternalDNS Operator take into account node roles when creating ExternalDNS Deployments (ie only schedule to control-plane nodes on OpenShift)?
+1. Will the initial ExternalDNS Operator be available via OLM, or the core OCP product, in OCP 4.9? (TODO: Find prior slack discussions around transitioning an operator from OLM to Core OCP)
```

Comment:
is there a way to node select or filter where the external dns controller runs?

Reply:
if delivered in core product, please clarify if its a net new second level operator, or delivered by an existing.

Reply:
related to core product as well, understanding what happens if this operator does not have valid credentials for a given provider with respect to cluster operator error reporting status would be worth calling out.

Reply:
Mentioned adding a node selector field to the operator CRD in the Proposal section.
Also mentioned a potential status condition for the CRD for the case you described (that could be bubbled up to the cluster operator).

Reply:
> if delivered in core product, please clarify if its a net new second level operator, or delivered by an existing
Could you elaborate on what you mean here? I'm not sure if I follow what the question is here.